src/expr.rs

```diff
@@ -2101,6 +2101,25 @@ pub(crate) fn rewrite_assign_rhs_expr<R: Rewrite>(
     rhs_kind: &RhsAssignKind<'_>,
     rhs_tactics: RhsTactics,
 ) -> Option<String> {
+    let get_lhs_last_line_shape = || {
+        let last_line = lhs.rsplitn(2, "\n").next().unwrap_or_default();
+        let tab_spaces = context.config.tab_spaces();
+        let new_shape = shape
+            .block_indent(tab_spaces)
+            .saturating_sub_width(tab_spaces);
```

Comment:
Can you explain the call to `.saturating_sub_width(tab_spaces)`. I'm not sure if that's necessary.

Reply:
Without saturating_sub_width, 
```rust
impl SomeType {
    fn method(&mut self) {
        self.array[array_index as usize]
            .as_mut()
            .expect("thing must exist")
            .extra_info =
            long_long_long_long_long_long_long_long_long_long_long_long_long_long_long;
    }
}
```
is formatted to
```rust
impl SomeType {
    fn method(&mut self) {
        self.array[array_index as usize]
            .as_mut()
            .expect("thing must exist")
            .extra_info = long_long_long_long_long_long_long_long_long_long_long_long_long_long_long;
    }
}
```
rather than
```rust
impl SomeType {
    fn method(&mut self) {
        self.array[array_index as usize]
            .as_mut()
            .expect("thing must exist")
            .extra_info =
                long_long_long_long_long_long_long_long_long_long_long_long_long_long_long;
    }
}
```

I think the remaining width is not accounted for properly if I only use block_indent

Reply:
Could you explain why we're using `tab_spaces`? What happens if we don't use `.saturating_sub_width(tab_spaces)`, but the rhs was even longer? Does it wrap correctly then?

Part of me wonders if the `shape` is off to begin with.

Reply:
I am using tab_spaces because the new shape is shifted to the right by exactly tab_spaces, so the available width should shrink by the same amount

Reply:
the shape that was used before adding extra block_indent is

```
old shape = Shape { width: 91, indent: Indent { block_indent: 8, alignment: 0 }, offset: 0 }
```

and the new shape with block_indent but without saturating_sub_width is
```
    let new_shape = shape
        .block_indent(context.config.tab_spaces());
```
```
new shape = Shape { width: 91, indent: Indent { block_indent: 12, alignment: 0 }, offset: 0
```

Reply:
ah...it's because the new shape is also the shape of the last line of lhs

Reply:
@ytmimi I think the above should answer your last question in this PR. Can you take a look?

Reply:
Thanks for explanation. I didn't have time to revisit this PR over the weekend, but I plan on giving this another look later this week